prov/psm/src/psmx_init.c

```diff
@@ -199,6 +199,52 @@ static int psmx_getinfo(uint32_t version, const char *node, const char *service,
 
 	*info = NULL;
 
+	/* Perform some quick check first to avoid unnecessary operations */
+	if (hints) {
+		if (hints->fabric_attr && hints->fabric_attr->name &&
+		    strcmp(hints->fabric_attr->name, PSMX_FABRIC_NAME)) {
+			FI_INFO(&psmx_prov, FI_LOG_CORE,
+				"hints->fabric_name=%s, supported=%s\n",
+				hints->fabric_attr->name, PSMX_FABRIC_NAME);
+			goto err_out;
+		}
+
+		if (hints->domain_attr && hints->domain_attr->name &&
+		    strcmp(hints->domain_attr->name, PSMX_DOMAIN_NAME)) {
+			FI_INFO(&psmx_prov, FI_LOG_CORE,
+				"hints->domain_name=%s, supported=%s\n",
+				hints->domain_attr->name, PSMX_DOMAIN_NAME);
+			goto err_out;
+		}
+
+		if (hints->ep_attr) {
+			switch (hints->ep_attr->type) {
+			case FI_EP_UNSPEC:
+			case FI_EP_DGRAM:
+			case FI_EP_RDM:
+				break;
+			default:
+				FI_INFO(&psmx_prov, FI_LOG_CORE,
+					"hints->ep_attr->type=%d, supported=%d,%d,%d.\n",
+					hints->ep_attr->type, FI_EP_UNSPEC,
+					FI_EP_DGRAM, FI_EP_RDM);
+				goto err_out;
+			}
+		}
+
+		if ((hints->caps & PSMX_CAPS) != hints->caps &&
+		    (hints->caps & PSMX_CAPS2) != hints->caps) {
+			uint64_t psmx_caps = PSMX_CAPS;
+			uint64_t psmx_caps2 = PSMX_CAPS2;
+			FI_INFO(&psmx_prov, FI_LOG_CORE, "hints->caps: %s,\n"
+				"supported = %s and %s.\n",
+				fi_tostr(&hints->caps, FI_TYPE_CAPS),
+				fi_tostr(&psmx_caps, FI_TYPE_CAPS),
+				fi_tostr(&psmx_caps2, FI_TYPE_CAPS));
```

Comment:
fi_tostr() uses a single internal buffer.  I'm not sure this will work without calling FI_INFO once per use of fi_tostr().  You may want to look at ofi_util.h, which has some macros to help with this.  See FI_INFO_MODE, for example.

Reply:
I remember seeing some weird output before but didn't look into the details. This is probably the reason. Will look into that.